### DISCUSSION TITLE
Limitation on returned stdout and stderr values from Cloud Functions

### DISCUSSION THREAD

**Discussion by arcanechimp** [appwrite/appwrite#3253] [Ideas]
Is this env variable mentioned in PR https://github.com/appwrite/appwrite/pull/1336 still going to be added?

I would like to use the Cloud Functions as a way to access data from a seperate cache and database whilst using Appwrite's authentication, and being able to change the max value would be very helpful to achieving this.

If the reason for the limitation is imposed to fit within the MariaDB string field size constraints, could the value just be truncated for storage in the logs, and the limit be removed entirely for the actual execution object returned from the function?

**Comment by stnguyen90** [comment]
The limit is currently [1MB](https://github.com/appwrite/appwrite/blob/f9056d1dcae27563b40b80f5c65908260f17d39e/app/executor.php#L306). Returning more than that in one response isn't really good practice.

**Reply by arcanechimp** [reply]
Thanks for your reply. 

I can see from the Function docs on the ExecutionObject, that the stdout and stderr strings are what is capped at 4000 chars, but a response string is where the actual data returned should be. 

However, no response string is being returned on the ExecutionObject when I call my NodeJS function, only the stdout and stderr strings. I can see my data being truncated in the stdout string, so the function appears to be running ok. This is why I was confused about the limitations on these strings. 

Is there any reason why the server would return an ExecutionObject without a response string, if the stdout is returning data?

**Reply by stnguyen90** [reply]
The code I linked was the latest version of Appwrite (0.14.2). You're probably on an older version. The cap was increased and stdout was renamed to response for clarity.

**Comment by arcanechimp** [comment]
Thank you! Updating to 14.2 solved the issue. Your help is much appreciated.
